### ISSUE TITLE
Create Default Error Page on Login (Iridium Core Server)

### ISSUE THREAD

**Issue by joshfischer1108** [IridiumIdentity/iridium#47]
**Is your feature request related to a problem? Please describe.**
When client errors happen during the login process, It would be useful to display a page that displays the error in a more human-friendly way that a JSON response

**Describe the solution you'd like**
We should create an error handling page to report errors out in a way that  will
1. Give the user some information of the error
2. Securely give the user information. (do not spill secrets, allow people to hit the database looking for users, etc. )

**Describe alternatives you've considered**
N/A

**Additional context**
The current view when errors occur is attached.
<img width="1675" alt="Screenshot 2023-06-04 at 3 44 48 PM" src="https://github.com/IridiumIdentity/iridium/assets/5785700/48f87d6f-2189-4858-82b1-46582f7cb218">

**Comment by timgrein** [comment]
Happy to look into that one.

Just some clarifying questions:
- This error page should live inside the `iridium-ui`, right? (not a default page inside the `ngx-iridium-client`; just asking as it also has a visible callback component)
- Do the backend services/controllers already provide all error messages needed (f.e. "user XY not registered") or is there some additional work to do?

**Comment by joshfischer1108** [comment]
Thanks for asking, Tim.  This error is related to the  `iridium-core-server` module.  The config is located [here](https://github.com/IridiumIdentity/iridium/blob/653dac1e05d33ea1fe561ba2fd190b2c0a547f16/iridium-core-server/src/main/java/software/iridium/api/Iridium.java#L48) and the templates are located [here](https://github.com/IridiumIdentity/iridium/tree/main/iridium-core-server/src/main/resources/templates).

Recreating this issue isn't as easy as I thought it would be.  You have to stand up the database, initialize it, run the API and hook up the management UI. I'll add the steps to create from scratch.  As the project grows and matures, we can make it easier.  But here we are for now 😬 .

* You'll need to run the following commands from the root directory of the project. 

### Stand up the database and mock smtp server
```
$  docker compose -f tools/schedulers/compose/local-dev-compose.yml up -d
```

* Now you can install the Iridium CLI somewhere and untar it, cd into the directory and run the init command.  This does not need to be in the root folder.
```
$ wget https://github.com/IridiumIdentity/iridium/releases/download/v0.1.4/iridium-0.1.4-bin.tar.gz
$ tar -xzf iridium-0.1.4-bin.tar.gz
$ cd iridium-0.1.4-bin
$ ./bin/iridium  init \
--host=localhost \
--port=3306 \
--user=root \
--password \
--admin-password \
--admin-email=you@somewhere.com
```
### The CLI tool should prompt you for the database password.  You can enter `iridium`.
```
$ Enter value for --password (the database password): 
```
### Next you'll need to enter the password for the admin user.  In this case it doesn't matter what you put because we are going to force an error by giving an incorrect login.  This _would_ be the password to accompany the admin email you provided with the cli arg `--admin-email=you@somewhere.com` if you were trying to do this correctly.  But since we are not, you can enter anything you like. 
```
$ Enter value for --admin-password (the admin password):
```
Finally you should see the following message if the CLI runs successfully.
```
16:29:38.758 [main] INFO  software.iridium.cli.command.InitCommand - ########################################
16:29:38.758 [main] INFO  software.iridium.cli.command.InitCommand - Database successfully initialized
16:29:38.758 [main] INFO  software.iridium.cli.command.InitCommand - ########################################
16:29:38.758 [main] INFO  software.iridium.cli.command.InitCommand - Iridium Management Application ID is f52e6b29d66761003f9711c94dc02bc8
16:29:38.758 [main] INFO  software.iridium.cli.command.InitCommand - Be sure to place this in the appropriate environment.ts file
```
You'll want to keep the generated Iridium Management APP ID for a later step. In this example, the value is `f52e6b29d66761003f9711c94dc02bc8`.  It'll need to replace the value `YOUR_CLIENT_ID` [here](https://github.com/IridiumIdentity/iridium/blob/43dba9864df7942117647207513b24ae75cac948/iridium-angular-client/projects/iridium-ui/src/environments/environment.ts#L8)

After the database is initialized you'll need to go back to the root folder of the project

* NOTE! You'll need to run the following commands from the root directory of the project. 

### Move properties into the resources folder
```
$ mv conf/application.properties  iridium-core-server/src/main/resources/
```
### Build the project
```
$ mvn clean package
```

### Run Iridium Core Server
```
$  java -jar iridium-core-server/target/iridium-core-server-0.1.4.jar
```

### Start the management UI
**Don't forget to add the generated application ID in the `environment.ts` file [here](https://github.com/IridiumIdentity/iridium/blob/43dba9864df7942117647207513b24ae75cac948/iridium-angular-client/projects/iridium-ui/src/environments/environment.ts#L8) before you complete the next step. 

```
$ cd iridium-angular-client && ng build ngx-iridium-client && ng serve
```

Then you visit the management UI in the browser at `localhost:4200` and attempt to log in with an incorrect username and password combination.  From here you should see an error in your browser that is plain JSON.  The project needs a way to catch these errors and display the error on a template that matches the style of the other templates in the project.     I'll attach a short video to demonstrate the error that comes back which is raw JSON.  


Again, this is a hefty set of instructions to reproduce an error.

**Comment by joshfischer1108** [comment]
https://github.com/IridiumIdentity/iridium/assets/5785700/100583d4-7a56-4ac7-bf4a-94092a6cc7dd

**Comment by timgrein** [comment]
Makes sense, thank you for the detailed explanation! I can still look into it, if no one else is working on it :)

**Comment by joshfischer1108** [comment]
It's yours if you want it 😄 .  If so, assign the issue to yourself, please.

**Comment by joshfischer1108** [comment]
resolved with #86
